### ISSUE TITLE
Problem with “occ files:scan –all –repair”: undefined index

### ISSUE THREAD

**Issue by j-snd** [owncloud/core#30291]
During upgrade from 10.0.4 to 10.0.5 I was asked to run "occ files:scan --all --repair". When I did, the following messages appeared:

```
Exception during scan: ErrorException: Undefined index: size
#0 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(420): OCA\Files\Command\Scan->exceptionErrorHandler(8, 'Undefined index...', '/var/www/ownclo...', 420, Array)                                                                         
#1 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(381): OC\Files\Cache\Scanner->handleChildren('files_encryptio...', true, 3, '280757', true, 0)         
#2 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '280757', true)              
#3 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '280120', true)              
#4 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '265504', true)              
#5 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '259131', true)              
#6 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '257434', true)              
#7 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '233361', true)              
#8 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '185633', true)              
#9 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(384): OC\Files\Cache\Scanner->scanChildren('files_encryptio...', true, 3, '10104', true)               
#10 /var/www/owncloud/lib/private/Files/Cache/Scanner.php(315): OC\Files\Cache\Scanner->scanChildren('', true, 3, '12', true)                                   
#11 /var/www/owncloud/lib/private/Files/Utils/Scanner.php(238): OC\Files\Cache\Scanner->scan('', true, 3)                                                       
#12 /var/www/owncloud/apps/files/lib/Command/Scan.php(224): OC\Files\Utils\Scanner->scan('/schneider', false)                                                   
#13 /var/www/owncloud/apps/files/lib/Command/Scan.php(310): OCA\Files\Command\Scan->scanFiles('schneider', '/schneider', false, Object(Symfony\Component\Console\Output\ConsoleOutput), false, false)                                           
#14 /var/www/owncloud/lib/composer/symfony/console/Command/Command.php(262): OCA\Files\Command\Scan->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))                         
#15 /var/www/owncloud/core/Command/Base.php(159): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))                                     
#16 /var/www/owncloud/lib/composer/symfony/console/Application.php(826): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/owncloud/lib/composer/symfony/console/Application.php(189): Symfony\Component\Console\Application->doRunCommand(Object(OCA\Files\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/owncloud/lib/composer/symfony/console/Application.php(120): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/owncloud/lib/private/Console/Application.php(161): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /var/www/owncloud/console.php(106): OC\Console\Application->run()
#21 /var/www/owncloud/occ(11): require_once('/var/www/ownclo...')
#22 {main}
```

What can I do about it? There seems to be an undefined index, ok, but where and why? What is the meaning of all these error messages?

The corresponding ownCloud instance was created with version 7 or before (not sure) and gradually upgraded since then till version 10.0.5 now.

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks the contributor most likely able to help you is @PVince81.

A possibly related issue is #26374: Just updated, slow sync - webdav file locked .

**Comment by j-snd** [comment]
Additional information eventually needed for solving the problem:
- Linux Debian Stretch
- apache 2.4.25
- php 7.0.27-0+deb9u1
- mariadb 10.1.26-0+deb9 (upgraded from mysql Jessie -> Stretch)
- containes nearly 200 GB of data
- many users with very different needs, only one affected by undefined index problem

**Comment by PVince81** [comment]
You can try with this patch https://github.com/owncloud/core/pull/30302

**Comment by j-snd** [comment]
Hello Vincent,

after applying your patch I have run the 'files:scan --all --repair' again. There are no error messages any more now. I have looked at some of the files and they seem to be ok. Since I could not extract any specific erroneous file from the error messages I don't know where to look further. Many error messages from ownCloud are for insiders only and are of minor use for normal users and admins.

Many thanks for your help!

Best regards
Jochen

**Comment by PVince81** [comment]
@j-snd this error message is not supposed to appear, it's just a side effect of a bug. So of course no one but insiders can understand it. Well, I only knew where to look because I investigated this bug myself, so no expectation there.

Your problem is likely related to https://github.com/owncloud/core/issues/28181.

There is another bug/regression where the actual error message isn't appearing properly. It is supposed to tell you that the file name in question has a different UTF-8 encoding (Mac NFD vs NFC). If you're using external storage you should turn on encoding compatibility mode on said storage.

**Comment by PifPof73** [comment]
I have the same problem on all users.
The patch and a new rescan did not help, no user can see any files, the menu buttons are not displayed (only the left site filters).
On Firefox I can see several errors listed in the console:
- TypeError: oc_appconfig.core is undefined  shareconfigmodel.js:24:4
- SyntaxError: unexpected token: ':'  app.js:132:5
- TypeError: oc_appconfig.filesReader is undefined  plugin.js:136:1
- TypeError: this.fileList is null app.js:132:5

Server:
- Ubuntu 16.04.3 LTS, Kernel 4.6.0-040600-generic
- PHP v7.0.27-1+ubuntu16.04.1+deb.sury.org+1
- Apache/2.4.18
- MySQL 5.7.21-0ubuntu0.16.04.1 (Ubuntu)
- ownCloud 10.0.6 (stable) 

Installed plugins:
> Enabled:
>   - activity: 2.3.6
>   - checksum: 0.3.5
>   - comments: 0.3.0
>   - configreport: 0.1.1
>   - customgroups: 0.3.6
>   - dav: 0.3.2
>   - encryption: 1.3.1
>   - federatedfilesharing: 0.3.1
>   - federation: 0.1.0
>   - files: 1.5.1
>   - files_antivirus: 0.11.2
>   - files_clipboard: 0.7.1
>   - files_external: 0.7.1
>   - files_reader: 1.2.1
>   - files_sharing: 0.10.1
>   - files_texteditor: 2.2.1
>   - files_trashbin: 0.9.1
>   - files_versions: 1.3.0
>   - files_videoplayer: 0.9.8
>   - firstrunwizard: 1.1
>   - gallery: 16.0.2
>   - gpxpod: 2.2.2
>   - impersonate: 0.1.2
>   - market: 0.2.3
>   - notifications: 0.3.2
>   - ownbackup: 17.7.0
>   - provisioning_api: 0.5.0
>   - systemtags: 0.3.0
>   - twofactor_totp: 0.4.3
>   - updatenotification: 0.2.1
>   - user_external: 0.4
>   - user_ldap: 0.10.0
> Disabled:
>   - guests
>   - richdocuments
>   - templateeditor
>   - theme-example

**Comment by PVince81** [comment]
@PifPof73 your problem looks like you didn't properly update the source code of ownCloud. Please check integrity check for errors. Best is to replace the source code with the tarball contents again

**Comment by PifPof73** [comment]
- The integrity check says the file "lib/private/Files/Storage/Wrapper/Checksum.php" is not correct, which is because I patched it.
- The ca-bundle.crt is not correct, because I added the crt of a collabora docker image

The rest seems to be ok.

I don't want to work with the source at all. Could I reinstall the owncloud-files package from the debian repository?

**Comment by PifPof73** [comment]
I reinstalled the package with
`apt-get install --reinstall owncloud-files`
Then I restarted the itegrity check
`sudo -u www-data php occ integrity:check-core`
And restarted the apache
`service apache2 restart`
I got no errors, but the files app is still not working.

**Comment by PifPof73** [comment]
Hi,
I finally found the reason for my problems.
- I did a complete remove of owncloud and reinstalled it. 
- Than I copied my config from the backup back.
- Restarted apache, everything worked (also the files \o/ ).
- I started to reinstall the apps from the market and testet the result
- If I install the fiels_reader (read ebooks) my installtion brakes
- it is approved by activating / deactivating the app

So this issue has to be moved to the developers of that app. Sorry for disturbing, but it is sometimes hard to get the details as "normal" sys administrator.
Thx,
PifPof73
